### DISCUSSION TITLE
引用博士论文时，年的位置应当在大学名称之后。

### DISCUSSION THREAD

**Discussion by earthmoon** [hushidong/biblatex-gb7714-2015#165] [Q&A]
您好，

我使用[武汉大学模板](https://github.com/whutug/whu-thesis)，在引用博士论文时，**年**的位置不对。可否帮我查找一下原因。

> @phdthesis{RN998,
> 	author = {曹洪},
> 	title = {博士论文名字},
> 	institution= {大学名称},
> 	type = {Thesis},
> 	abstract = {这是摘要},
> 	keywords = {关键词1
> 关键词2
> 关键词3
> 关键词4},
> 	year = {2023},
> 	key = {cao2hong2},
> }

**年**的位置太靠前了，按照国标，**年**的位置应该在大学名称之后，但是实际在姓名之后。
谢谢！

**Comment by hushidong** [comment]
年不会丢啊，主要是学校没有吧？
把university改为school或institution就好了。

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: earthmoon ***@***.***>
Sent: Thursday, May 4, 2023 11:48:45 AM
To: hushidong/biblatex-gb7714-2015 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [hushidong/biblatex-gb7714-2015] 请教一个问题，这篇博士论文为何引用时没有年、大学名称？ (Discussion #165)


您好，

我使用武汉大学模板<https://github.com/whutug/whu-thesis>，在引用博士论文时，发现没有年、大学名称。可否帮我查找一下原因。

@PhDThesis<https://github.com/PhDThesis>{RN998,
author = {曹洪},
title = {博士论文名字},
university = {大学名称},
type = {Thesis},
abstract = {这是摘要},
keywords = {关键词1
关键词2
关键词3
关键词4},
year = {2023},
key = {cao2hong2},
}

谢谢！

―

**Reply by earthmoon** [reply]
> 年不会丢啊，主要是学校没有吧？ 把university改为school或institution就好了。 获取Outlook for Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: earthmoon ***@***.***> Sent: Thursday, May 4, 2023 11:48:45 AM To: hushidong/biblatex-gb7714-2015 ***@***.***> Cc: Subscribed ***@***.***> Subject: [hushidong/biblatex-gb7714-2015] 请教一个问题，这篇博士论文为何引用时没有年、大学名称？ (Discussion #165) 您好， 我使用武汉大学模板<[https://github.com/whutug/whu-thesis>，在引用博士论文时，发现没有年、大学名称。可否帮我查找一下原因。](https://github.com/whutug/whu-thesis>%EF%BC%8C%E5%9C%A8%E5%BC%95%E7%94%A8%E5%8D%9A%E5%A3%AB%E8%AE%BA%E6%96%87%E6%97%B6%EF%BC%8C%E5%8F%91%E7%8E%B0%E6%B2%A1%E6%9C%89%E5%B9%B4%E3%80%81%E5%A4%A7%E5%AD%A6%E5%90%8D%E7%A7%B0%E3%80%82%E5%8F%AF%E5%90%A6%E5%B8%AE%E6%88%91%E6%9F%A5%E6%89%BE%E4%B8%80%E4%B8%8B%E5%8E%9F%E5%9B%A0%E3%80%82) @PhDThesis<[https://github.com/PhDThesis>{RN998](https://github.com/PhDThesis>%7BRN998), author = {曹洪}, title = {博士论文名字}, university = {大学名称}, type = {Thesis}, abstract = {这是摘要}, keywords = {关键词1 关键词2 关键词3 关键词4}, year = {2023}, key = {cao2hong2}, } 谢谢！ ― Reply to this email directly, view it on GitHub<#165>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE3ZZI74FPFLXGA3ZBN4S63XEMRJ3ANCNFSM6AAAAAAXVHHJK4>. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

谢谢！生成参考文献年的位置有问题：年的位置太靠前了，按照国标，年的位置应该在大学名称之后，但是实际在姓名之后。

**Comment by hushidong** [comment]
你用的不是作者-年份制么？正常应该就在作者后面的，不过也可以改，你看一下mergedate选项。

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by earthmoon** [reply]
> 你用的不是作者-年份制么？正常应该就在作者后面的，不过也可以改，你看一下mergedate选项。 获取Outlook for Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: earthmoon ***@***.***> Sent: Friday, May 5, 2023 9:36:50 AM To: hushidong/biblatex-gb7714-2015 ***@***.***> Cc: hzzmail ***@***.***>; Comment ***@***.***> Subject: Re: [hushidong/biblatex-gb7714-2015] 引用博士论文时，年的位置应当在大学名称之后。 (Discussion #165) 年不会丢啊，主要是学校没有吧？ 把university改为school或institution就好了。 获取Outlook for Androidhttps://aka.ms/AAb9ysg …
> ________________________________ From: earthmoon @.> Sent: Thursday, May 4, 2023 11:48:45 AM To: hushidong/biblatex-gb7714-2015 @.> Cc: Subscribed @.> Subject: [hushidong/biblatex-gb7714-2015] 请教一个问题，这篇博士论文为何引用时没有年、大学名称？ (Discussion #165<#165>) 您好， 我使用武汉大学模板<[https://github.com/whutug/whu-thesis>，在引用博士论文时，发现没有年、大学名称。可否帮我查找一下原因。<https://github.com/whutug/whu-thesis%3E%EF%BC%8C%E5%9C%A8%E5%BC%95%E7%94%A8%E5%8D%9A%E5%A3%AB%E8%AE%BA%E6%96%87%E6%97%B6%EF%BC%8C%E5%8F%91%E7%8E%B0%E6%B2%A1%E6%9C%89%E5%B9%B4%E3%80%81%E5%A4%A7%E5%AD%A6%E5%90%8D%E7%A7%B0%E3%80%82%E5%8F%AF%E5%90%A6%E5%B8%AE%E6%88%91%E6%9F%A5%E6%89%BE%E4%B8%80%E4%B8%8B%E5%8E%9F%E5%9B%A0%E3%80%82](https://github.com/whutug/whu-thesis>%EF%BC%8C%E5%9C%A8%E5%BC%95%E7%94%A8%E5%8D%9A%E5%A3%AB%E8%AE%BA%E6%96%87%E6%97%B6%EF%BC%8C%E5%8F%91%E7%8E%B0%E6%B2%A1%E6%9C%89%E5%B9%B4%E3%80%81%E5%A4%A7%E5%AD%A6%E5%90%8D%E7%A7%B0%E3%80%82%E5%8F%AF%E5%90%A6%E5%B8%AE%E6%88%91%E6%9F%A5%E6%89%BE%E4%B8%80%E4%B8%8B%E5%8E%9F%E5%9B%A0%E3%80%82<https://github.com/whutug/whu-thesis%3E%EF%BC%8C%E5%9C%A8%E5%BC%95%E7%94%A8%E5%8D%9A%E5%A3%AB%E8%AE%BA%E6%96%87%E6%97%B6%EF%BC%8C%E5%8F%91%E7%8E%B0%E6%B2%A1%E6%9C%89%E5%B9%B4%E3%80%81%E5%A4%A7%E5%AD%A6%E5%90%8D%E7%A7%B0%E3%80%82%E5%8F%AF%E5%90%A6%E5%B8%AE%E6%88%91%E6%9F%A5%E6%89%BE%E4%B8%80%E4%B8%8B%E5%8E%9F%E5%9B%A0%E3%80%82)> @PhDThesis<[https://github.com/PhDThesis><https://github.com/PhDThesis>{RN998<https://github.com/PhDThesis%3E%7BRN998](https://github.com/PhDThesis><https://github.com/PhDThesis>%7BRN998<https://github.com/PhDThesis%3E%7BRN998)>, author = {曹洪}, title = {博士论文名字}, university = {大学名称}, type = {Thesis}, abstract = {这是摘要}, keywords = {关键词1 关键词2 关键词3 关键词4}, year = {2023}, key = {cao2hong2}, } 谢谢！ D Reply to this email directly, view it on GitHub<#165<#165>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AE3ZZI74FPFLXGA3ZBN4S63XEMRJ3ANCNFSM6AAAAAAXVHHJK4. You are receiving this because you are subscribed to this thread.Message ID: @.> 谢谢！生成参考文献年的位置有问题：年的位置太靠前了，按照国标，年的位置应该在大学名称之后，但是实际在姓名之后。 ― Reply to this email directly, view it on GitHub<[#165 (reply in thread)](https://github.com/hushidong/biblatex-gb7714-2015/discussions/165#discussioncomment-5811362)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE3ZZI3NTVLETJUIZKK4SB3XERKTFANCNFSM6AAAAAAXVHHJK4>. You are receiving this because you commented.Message ID: ***@***.***>

现在明白你说的意思了，你说的是对的。我误解了。谢谢！

**Comment by earthmoon** [comment]
比如

> PhDThesis{RN998,
> author = {曹洪},
> title = {博士论文名字},
> institution= {大学名称},
> type = {Thesis},
> abstract = {这是摘要},
> keywords = {关键词1
> 关键词2
> 关键词3
> 关键词4},
> year = {2023},
> key = {cao2hong2},
> }
> 

生成： 

> 曹洪, 2023. 博士论文名字[D]. 大学名称.

按照国标应该为：
>曹洪. 博士论文名字[D]. 大学名称, 2023.

这是国标中的截图：
![iamge](https://user-images.githubusercontent.com/45897471/236364224-d03eacb8-a4f3-41f5-8d8a-1802be3a1ef2.png)

谢谢！

**Reply by note286** [reply]
```latex
\documentclass{ctexart}
\usepackage[style=gb7714-2015]{biblatex}
\addbibresource{reference.bib}
\begin{document}
\cite{RN998}
\printbibliography
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/236366522-ccfdfe8d-6f9d-45ac-b061-d2deabd32f2e.png)

**Reply by earthmoon** [reply]
> ```
> \documentclass{ctexart}
> \usepackage[style=gb7714-2015]{biblatex}
> \addbibresource{reference.bib}
> \begin{document}
> \cite{RN998}
> \printbibliography
> \end{document}
> ```
> 
> ![image](https://user-images.githubusercontent.com/16207292/236366522-ccfdfe8d-6f9d-45ac-b061-d2deabd32f2e.png)

那这是武大模板的问题，我去那边反映。

**Comment by sikouhjw** [comment]
修正你的认知，你的认知是：
- 『年』在末尾

根据国标，顺序编码制才是如此。而武大硕博规定使用作者-年份制。根据 @hushidong 说的，在作者-年份制中，
> 正常应该就在作者后面的
```tex
\begin{filecontents}{xampl-zh.bib}
@phdthesis{zhanghesheng1998dizhilixue,
  author      = {张和生},
  school      = {太原理工大学},
  title       = {地质力学系统理论},
  year        = {1998},
  address     = {太原},
}
\end{filecontents}

\documentclass{ctexart}
\usepackage[backend=biber,style=gb7714-2015ay]{biblatex}
\addbibresource{xampl-zh.bib}
\begin{document}
\cite{zhanghesheng1998dizhilixue}
\printbibliography[heading=bibliography,title=参考文献]
\end{document}
```
![图片](https://user-images.githubusercontent.com/49770925/236367964-98f21876-9e23-467e-8c33-5bac444bc2f5.png)

你从武大模板提问至今为止，在 latex 领域没有进步的地方在于：
1. 提问不给完整信息
    - 武大模板有本硕博三种，我怎么知道你用的是哪一种？
    - 为什么提问不给 MWE，而是让回答者自己去做 MWE 来测试
    - 不给 MWE 会浪费彼此的时间，在沟通上
2. 喜欢刨根问底却喜欢提『容易自己找到答案』的问题[^1]
3. 喜欢自己乱改东西，拿乱改后出现的问题来问人[^2]
4. 没有经过大量的测试，形成自己的判断和理性认识就匆匆提问[^3]

[^1]: https://github.com/whutug/whu-thesis/discussions/136#discussioncomment-5792102
[^2]: https://github.com/whutug/whu-thesis/discussions/174
[^3]: https://github.com/whutug/whu-thesis/discussions/184

**Reply by earthmoon** [reply]
> 修正你的认知，你的认知是：
> 
> * 『年』在末尾
> 
> 根据国标，顺序编码制才是如此。而武大硕博规定使用作者-年份制。根据 @hushidong 说的，在作者-年份制中，
> 
> > 正常应该就在作者后面的
> 
> ```
> \begin{filecontents}{xampl-zh.bib}
> @phdthesis{zhanghesheng1998dizhilixue,
>   author      = {张和生},
>   school      = {太原理工大学},
>   title       = {地质力学系统理论},
>   year        = {1998},
>   address     = {太原},
> }
> \end{filecontents}
> 
> \documentclass{ctexart}
> \usepackage[backend=biber,style=gb7714-2015ay]{biblatex}
> \addbibresource{xampl-zh.bib}
> \begin{document}
> \cite{zhanghesheng1998dizhilixue}
> \printbibliography[heading=bibliography,title=参考文献]
> \end{document}
> ```
> 
> ![图片](https://user-images.githubusercontent.com/49770925/236367964-98f21876-9e23-467e-8c33-5bac444bc2f5.png)
> 
> 你从武大模板提问至今为止，在 latex 领域没有进步的地方在于：
> 
> 1. 提问不给完整信息
>    
>    * 武大模板有本硕博三种，我怎么知道你用的是哪一种？
>    * 为什么提问不给 MWE，而是让回答者自己去做 MWE 来测试
>    * 不给 MWE 会浪费彼此的时间，在沟通上
> 2. 喜欢刨根问底却喜欢提『容易自己找到答案』的问题[1](#user-content-fn-1-2be55810e2e7900b175db555b07db24c)
> 3. 喜欢自己乱改东西，拿乱改后出现的问题来问人[2](#user-content-fn-2-2be55810e2e7900b175db555b07db24c)
> 4. 没有经过大量的测试，形成自己的判断和理性认识就匆匆提问[3](#user-content-fn-3-2be55810e2e7900b175db555b07db24c)
> 
> ## Footnotes
> 1. https://github.com/whutug/whu-thesis/discussions/136#discussioncomment-5792102 [↩](#user-content-fnref-1-2be55810e2e7900b175db555b07db24c)
> 2. https://github.com/whutug/whu-thesis/discussions/174 [↩](#user-content-fnref-2-2be55810e2e7900b175db555b07db24c)
> 3. https://github.com/whutug/whu-thesis/discussions/184 [↩](#user-content-fnref-3-2be55810e2e7900b175db555b07db24c)

你说的是对的，下次不会了。
